fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -385,7 +384,9 @@ protected String buildWhereClause(String overrideType) {
                             break;
                         }
                         whereClauseSegment = whereClauseSegment.replaceAll(PARAMETER_TABLE_ALIAS + ".", "");
-                        whereClause.append(" WHERE ").append(whereClauseSegment).append(")");
+                        whereClause.append(" WHERE ").append(whereClauseSegment).append(") ");
+                        String tmpTableName = "TMP" + i;
```

Comment:
at this point in the code you know the tmpTableName do you know the SimpleName for the Resource Class? Location.class.getSimpleName

You can be explicit with the name LOCATION_1. 

I'm not sure that this 'optimizes' anything.  Rather, it may make debugging the sql easier.  so there is a tradeoff.

Reply:
sound good for me, actually, there resource type name can be override, I'm going to change it into overrideType instead, no cost